### ISSUE TITLE
Unseal GeoJson classes?

### ISSUE THREAD

**Issue by jaybo** [Azure/azure-cosmos-dotnet-v3#1980]
The GeoJson specification (https://tools.ietf.org/html/rfc7946) allows for extensions and additional foreign members: 

> **6.1.  Foreign Members**
> Members not described in this specification ("foreign members") MAY be used in a GeoJSON document.

But since all of the GeoJson classes in the Spatial namespace are marked sealed, subclassing is not available, and hence foreign members are not possible.

What is the justification for the sealed modifier and why not remove it to make the Spatial namespace classes usable for a larger audience?

One glaring example is the need to add **id** and other Cosmos related document properties.  Or is it assumed Spatial namespace classes will never be root documents?

**Comment by bchong95** [comment]
As the rfc mentions while geojsons MAY have foreign members, they are frowned upon and the RFC states that you lose interoperability if you have them.

As you have correctly brought up in a previous issue:

https://github.com/Azure/azure-cosmos-dotnet-v3/issues/1854

The cleaner way to achieve this is to use composition with feature and feature collection. 

For instead of:

```JSON
{
         "type": "Point",
         "coordinates": [100.0, 0.0],
         "some foriegn property A": "asdf"
}
``` 
 
You can model your object as:

```JSON
{
       "type": "Feature",
       "geometry": {
           "type": "Point",
         "coordinates": [100.0, 0.0],
       },
       "properties":{
           "A": "asdf"
        }
}
```

**Comment by jaybo** [comment]
Right, but CosmosDB requires that an "id" property at the root document level.  Hence, none of the Spatial data types can exist at the root document level in a container at present.  Simply making these classes unsealed allows for that possibility.

**Comment by jackbond** [comment]
Unless I'm mistaken, future versions of the Cosmos libraries will not contain spatial classes at all. Instead, they'll be imported from a common Azure NuGet library. See...

https://github.com/Azure/azure-cosmos-dotnet-v3/pull/1535

I don't know for sure, but this MIGHT be it.

https://www.nuget.org/packages/Microsoft.Azure.Core.Spatial/1.0.0-beta.1

Bottom line, you might have better luck hitting that team up for the types of changes you are looking for.
